lib/class-wp-theme-json-resolver.php

```diff
@@ -482,10 +482,37 @@ public static function get_user_custom_post_type_id() {
 	 */
 	public static function theme_has_support() {
 		if ( ! isset( self::$theme_has_support ) ) {
-			self::$theme_has_support = is_readable( locate_template( 'experimental-theme.json' ) );
+			self::$theme_has_support = !! self::get_file_path_from_theme( 'experimental-theme.json' );
```

Comment:
Maybe we can cast the type to make it more readable.

```suggestion
			self::$theme_has_support = (bool) self::get_file_path_from_theme( 'experimental-theme.json' );
```

Reply:
I like that: https://github.com/WordPress/gutenberg/pull/28942/commits/23e0a78c8c7d5049db3d9b166fba94bf620d9896